scripts/first_boot_partitioning

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+# create a partition scheme on first bootup that looks like this:
+# - boot partition, original size
+# - system partition #1, 10G
+# - system partition #2, 10G
+# - state & /home partition, 10G
+# this scheme permits us to install new operating systems over the air and in the future
+# simplify upgrades. during time of implementation, we have not roadmapped this OTA functionality yet.
+# ref: https://github.com/micro-nova/AmpliPi/issues/489
+#
+# A lot of this script was inspired by (or copied from) Raspberry Pi's init_resize.sh; cheers yall.
+# ref: https://github.com/RPi-Distro/raspi-config/blob/408bde537671de6df2d9b91564e67132f98ffa71/usr/lib/raspi-config/init_resize.sh
+# ref: https://github.com/RPi-Distro/raspi-config/blob/408bde537671de6df2d9b91564e67132f98ffa71/LICENSE
+
+echo "starting first boot partitioning tasks..."
+
+# bail if any command uses an unset variable
+set -u
+
+# spit out the execution of each line; in case we do bail we know where it failed
+set -x
+
+# kernel panic messages will quickly overwhelm the screen if we don't trap the exit during
+# potential troubleshooting
+trap 'sleep 10' EXIT
```

Comment:
gentle balance here; 10s feels fine enough to wait patiently during first boot of the appliance, but also enough time to snap a picture on your cellphone during a broken install (speaking from today's experience :woozy_face: )